Jenkinsfile

```diff
@@ -67,18 +81,65 @@ node('mac-mini-ios') {
                     timeout(10) {
                         sh '''#!/bin/bash -l
                             set -e
-                            xcodebuild -workspace Client.xcworkspace -scheme "Fennec" -sdk iphonesimulator -destination "platform=iOS Simulator,OS=11.2,id=185B34BB-DCB8-4A17-BDCA-843086B67193" ONLY_ACTIVE_ARCH=NO -derivedDataPath clean build
+                            xcodebuild -workspace Client.xcworkspace -scheme "Fennec" -sdk iphonesimulator -destination "platform=iOS Simulator,OS=11.2,id=185B34BB-DCB8-4A17-BDCA-843086B67193" ONLY_ACTIVE_ARCH=NO -derivedDataPath clean build test
                         '''
                     }
                 }
+                stage('Setup Test Environment'){
+                    sh '''#!/bin/bash -l
+                        set -e
+                        npm install -g appium
+                        npm install -g wd
+                        appium &
+                        echo $! > appium.pid
+                        sleep 10
+                    '''
+                }
+                stage('Run Tests') {
+                    withEnv([
+                        'platformName=ios',
+                        'udid=185B34BB-DCB8-4A17-BDCA-843086B67193',
+                        'deviceName=iPhone 6',
+                        'platformVersion=11.2',
+                        'MODULE=testSmoke',
+                        'TEST=SmokeTest',
+                        'bundleID=com.cliqz.ios.newCliqz'
+                        ]) {
+                        timeout(60) {
+                            sh '''#!/bin/bash -l
+                                set -x
+                                set -e
+                                python autobots/testRunner.py
+                            '''
+                        }
+                    }
+                }
             }
             catch(all) {
                 jobStatus = 'FAIL'
             }
             finally {
+                stage('Upload Results') {
+                    try {
+                        archiveArtifacts allowEmptyArchive: true, artifacts: 'autobots/*.log'
+                        junit "autobots/test-reports/*.xml"
+                        zip archive: true, dir: 'autobots/screenshots', glob: '', zipFile: 'autobots/screenshots.zip'
+                    } catch (e) {
+                        // no screenshots, no problem
```

Comment:
have seen this somewhere before? can we extract something to our `jenkins-pipeline-library` ? /cc @sharath-cliqz